src/web/server/document.tsx

```diff
@@ -109,27 +106,31 @@ export const document = ({ data }: Props): string => {
 	// Pass the array of extracted (read: built with addChunk) scripts to
 	// generatedScriptTags so that we can build a script tag array of
 	// modern and legacy scripts.
-	const preAssets = loadableExtractor
+	const arrayOfLoadableScriptObjects: {
+		src: string;
+		module: boolean;
+	}[] = loadableExtractor
 		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 		// @ts-ignore
-		.getPreAssets(); // PreAssets is *undocumented* and not in TS types. It returns the webpack asset for each script.
-	// Pre assets returns an array of objects structured as:
-	// {
-	//     filename: 'elements-RichLinkComponent.js',
-	//     scriptType: 'script',
-	//     chunk: 'elements-RichLinkComponent',
-	//     url: '/elements-RichLinkComponent.js',
-	//     path: '/Users/gareth_trufitt/code/dotcom-rendering/dist/elements-RichLinkComponent.js',
-	//     type: 'mainAsset',
-	//     linkType: 'preload'
-	// }
-
-	const loadableScripts = generateScriptTags(
-		(preAssets || []).map((script: { url: string }) => ({
-			src: `${script.url}`,
-			module: true,
-		})),
-	);
+		.getPreAssets() // PreAssets is *undocumented* and not in TS types. It returns the webpack asset for each script.
+		// Pre assets returns an array of objects structured as:
+		// {
+		//     filename: 'elements-RichLinkComponent.js',
+		//     scriptType: 'script',
+		//     chunk: 'elements-RichLinkComponent',
+		//     url: '/elements-RichLinkComponent.js',
+		//     path: '/Users/gareth_trufitt/code/dotcom-rendering/dist/elements-RichLinkComponent.js',
+		//     type: 'mainAsset',
+		//     linkType: 'preload'
+		// }
+		.reduce(
```

Comment:
Whenever I see reduce my brain is like whelp 🤷‍♂️ , something with adding stuff? I know that there's an accumulator and that we're interating but I trip up because it's not explicitly written out whats happening and because I'm not used to using this function I really have to _think_ to see it, it's like javascript written in French for me, I _can_ read it but I'm left wondering if I missed some subtleties...

I think that this is probably taking one array, iterating it and producing another one after doing a transformation? Maybe map works?

On the other hand, this might be a perfectly valid use case for reduce and maybe I need to study it more!

Reply:
Yea, valid. I don't think I can use map because I need to `merge` the arrays, but maybe it would help with a simple `let` and `forEach`. Again, I'll give it a go after this PR with tests and a refactoring PR.